frontend/packages/gitops-plugin/src/components/list/GitOpsList.tsx

```diff
@@ -1,41 +1,49 @@
 import * as React from 'react';
-import * as _ from 'lodash';
 import { useTranslation } from 'react-i18next';
-import GitOpsListItem from './GitOpsListItem';
-import { Stack, StackItem, Split, SplitItem } from '@patternfly/react-core';
-import { K8sResourceKind } from '@console/internal/module/k8s';
 import { GitOpsAppGroupData } from '../utils/gitops-types';
+import GitOpsTableRow from './GitOpsTableRow';
+import GitOpsTableHeader from './GitOpsTableHeader';
 import GitOpsEmptyState from '../GitOpsEmptyState';
+import { Table, TextFilter } from '@console/internal/components/factory';
+import { fuzzyCaseInsensitive } from '@console/internal/components/factory/table-filters';
 import './GitOpsList.scss';
 
 interface GitOpsListProps {
   appGroups: GitOpsAppGroupData[];
   emptyStateMsg: string;
-  argocdLink?: K8sResourceKind;
 }
 
-const GitOpsList: React.FC<GitOpsListProps> = ({ appGroups, emptyStateMsg, argocdLink }) => {
+const GitOpsList: React.FC<GitOpsListProps> = ({ appGroups, emptyStateMsg }) => {
   const { t } = useTranslation();
+  const [textFilter, setTextFilter] = React.useState('');
+
+  const visibleItems = appGroups?.filter(({ name }) => {
+    return (
+      fuzzyCaseInsensitive(textFilter, name) ||
+      name.toLowerCase().indexOf(textFilter.toLowerCase()) !== -1
```

Comment:
Fuzzy should cover an indexOf check.

Reply:
Sorry not sure I understand this one, can you elaborate? Or point to somewhere this is happening? I tried to look at other implementations and I don't think I'm seeing this unless I'm just missing it

Reply:
`fuzzyCaseInsensitive` covers the "indexOf" case… so you’ll never end up with `false || true` in this statement… you’ll either get `true || true`, `true || false` or naturally `false || false` because Fuzzy is a superset calculation of the indexOf string.

Reply:
ohhh I see, okay. I saw it with the `indexOf` check in other places so I thought it might be necessary. I'll remove it, thanks for the explanation!